tap_hubspot/__init__.py

```diff
@@ -307,8 +322,70 @@ def lift_properties_and_versions(record):
             record['properties_versions'] += versions
     return record
 
+def post_search_endpoint(url, data, params=None):
+    params = params or {}
+    hapikey = CONFIG['hapikey']
+    if hapikey is None:
+        if CONFIG['token_expires'] is None or CONFIG['token_expires'] < datetime.datetime.utcnow():
+            acquire_access_token_from_refresh_token()
```

Comment:
I slightly remember discussing this in a previous PR... but why not make the code simply:
`headers = {'Authorization': 'Bearer {}'.format(access_token())}`

And then pushing this logic of `if CONFIG['token_expires'] is None or CONFIG['token_expires'] < datetime.datetime.utcnow():` into an access_token method which would take care of it? It feels like a leaky abstraction right now that seems unnecessary.

Reply:
I moved this into a function we use in both functions now